test/e2e/mtu_test.go

```diff
@@ -0,0 +1,211 @@
+// Copyright 2024 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"testing"
+
+	agentconfig "antrea.io/antrea/pkg/config/agent"
+)
+
+func TestPodMTU(t *testing.T) {
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	t.Run("testPodMTUWithGeneveTunnel", func(t *testing.T) {
+		testPodMTUWithGeneveTunnel(t, data)
+	})
+	t.Run("testPodMTUWithVXLANTunnel", func(t *testing.T) {
+		testPodMTUWithVXLANTunnel(t, data)
+	})
+	t.Run("testPodMTUWithGRETunnel", func(t *testing.T) {
+		testPodMTUWithGRETunnel(t, data)
+	})
+	t.Run("testPodMTUWithWireGuard", func(t *testing.T) {
+		testPodMTUWithWireGuard(t, data)
+	})
+}
+
+func testPodMTUWithGeneveTunnel(t *testing.T, data *TestData) {
+	var previousTunnelType string
+	ac := func(config *agentconfig.AgentConfig) {
+		previousTunnelType = config.TunnelType
+		config.TunnelType = "geneve"
+	}
+	if err := data.mutateAntreaConfigMap(nil, ac, false, true); err != nil {
+		t.Fatalf("Failed to enable geneve tunnel: %v", err)
+	}
+	defer func() {
+		ac := func(config *agentconfig.AgentConfig) {
+			config.TunnelType = previousTunnelType
+		}
+		if err := data.mutateAntreaConfigMap(nil, ac, false, true); err != nil {
+			t.Errorf("Failed to enable %s tunnel: %v", previousTunnelType, err)
+		}
+	}()
+
+	podName := "geneve"
+	err := data.createBusyboxPodOnNode(podName, data.testNamespace, workerNodeName(1), false)
+	if err != nil {
+		t.Fatalf("Failed to create busybox pod: %v", err)
+	}
+	defer deletePodWrapper(t, data, data.testNamespace, podName)
+	if err := data.podWaitForRunning(defaultTimeout, podName, data.testNamespace); err != nil {
+		t.Fatalf("Error when waiting for Pod %s: %v", podName, err)
+	}
+
+	podInterface, err := data.GetPodInterfaceName(workerNodeName(1), podName)
+	if err != nil {
+		t.Fatalf("Failed to get Pod interface: %v", err)
+	}
+	mtu, err := data.GetInterfaceMTU(workerNodeName(1), podInterface)
+	if err != nil {
+		t.Fatalf("Failed to get interface MTU: %v", err)
+	}
+	if mtu != 1450 {
```

Comment:
You defined a few MTU consts, maybe refer to them in mtu_test.go as well.

Reply:
Since you added traffic connection verification, maybe mtu_test.go is no longer needed since the MTU calculation can be validated in unit tests. @tnqn any thoughts?

Reply:
According to the discussion with Quan last review, the `mtu_test.go` could be removed cause we have verified the MTU in UT.